File: src/types/query.ts
export interface QueryRequest {
  connection_id: string;
  sql: string;
  database?: string;
  page?: number;
  page_size?: number;
}

export interface QueryResult {
  columns: string[];
  rows: Array<Array<any>>;
  total_rows: number;
  execution_time_ms: number;
}

export interface QueryHistoryEntry {
  id: string;
  connection_id: string;
  sql: string;
  executed_at: number;
  execution_time_ms: number;
  success: boolean;
  error_message?: string;
}

File: src-tauri/src/models/query.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct QueryRequest {
    pub connection_id: String,
    pub sql: String,
    pub database: Option<String>,
    pub page: Option<u32>,
    pub page_size: Option<u32>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct QueryResult {
    pub columns: Vec<String>,
    pub rows: Vec<Vec<serde_json::Value>>,
    pub total_rows: usize,
    pub execution_time_ms: u64,
}
